internal/reporter.go

```diff
@@ -77,8 +78,9 @@ func (reporter reporter) buildRequest(format string, body []byte) (*http.Request
 
 	req.Header.Set(contentType, octetStream)
 	req.Header.Set(contentEncoding, gzipFormat)
-	if len(reporter.token) > 0 {
```

Comment:
I'd like to play with this area a bit...

instead of `len` checks, I wonder if we can pull this out into the authorization service and do something like:

`authorizationService.AddHeaders(request)`